### PR TITLE
Graceful trigger fail

### PR DESCRIPTION
**Pull request by AndrewEQ**
Added graceful fail to trigger when its not rooted else it crashes the app with the exception.

### CODE REVIEW

File: Source/Fuse.Triggers/Trigger.uno [hunk 1]
```diff
@@ -341,8 +341,11 @@ namespace Fuse.Triggers
 		{
 			if (!_isStarted)
 			{
-				if (!IsRootingStarted)
-					throw new Exception( "Trigger started prior to being rooted: " + this );
+				if (!IsRootingStarted){
+					debug_log("Warning: Trigger.uno - Trigger started prior to being rooted: " + this);
+					return;
+				}
```
Comment (kusma) [inline Source/Fuse.Triggers/Trigger.uno]: Hmm, this just hides the symptoms... Why do you want to do that rather than fixing the actual problem?

Comment (mortend) [reply Source/Fuse.Triggers/Trigger.uno]: I think we should call `Fuse.Diagnostics.UserWarning(message, object)` to report the warning instead.

https://github.com/fuse-open/fuselibs/search?q=UserWarning&unscoped_q=UserWarning

Comment (AndrewEQ) [reply Source/Fuse.Triggers/Trigger.uno]: Tried fixing the problem but it randomly occurs, don't know why things are not rooted at the time of triggering the animation, nor how to make sure that it is rooted for sure before triggering the animation.

Comment (mortend) [reply Source/Fuse.Triggers/Trigger.uno]: @kusma I don't think this have to be a fatal error, so reporting the problem via the Diagnostics API is maybe better than crashing the app.

Comment (AndrewEQ) [reply Source/Fuse.Triggers/Trigger.uno]: ...switched it to Fuse.Diagnostics.UserError()

File: Source/Fuse.Triggers/Trigger.uno [hunk 2]
```diff
@@ -341,8 +341,11 @@ namespace Fuse.Triggers
 		{
 			if (!_isStarted)
 			{
-				if (!IsRootingStarted)
-					throw new Exception( "Trigger started prior to being rooted: " + this );
+				if (!IsRootingStarted){
+					Fuse.Diagnostics.UserError("Warning: Trigger.uno - Trigger started prior to being rooted: ", this );
```
Comment (mortend) [inline Source/Fuse.Triggers/Trigger.uno]:
```suggestion
					Fuse.Diagnostics.UserError("Trigger started prior to being rooted", this);
```

File: Source/Fuse.Triggers/Trigger.uno [hunk 3]
```diff
@@ -341,8 +341,11 @@ namespace Fuse.Triggers
 		{
 			if (!_isStarted)
 			{
-				if (!IsRootingStarted)
-					throw new Exception( "Trigger started prior to being rooted: " + this );
+				if (!IsRootingStarted){
```
Comment (mortend) [inline Source/Fuse.Triggers/Trigger.uno]: Code-style: `{` that should've been on its own line.

Comment (AndrewEQ) [reply Source/Fuse.Triggers/Trigger.uno]: sorted

**Pull request merged**